platform/security/idp/security-idp-plugin/security-idp-plugin-nameidpolicy/src/main/java/org/codice/ddf/security/idp/plugin/nameidpolicy/NameIdPolicyPresignPlugin.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.idp.plugin.nameidpolicy;
+
+import ddf.security.SubjectUtils;
+import ddf.security.samlp.SamlProtocol.Binding;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.security.idp.plugin.SamlPresignPlugin;
+import org.opensaml.core.xml.schema.XSString;
+import org.opensaml.saml.saml2.core.Attribute;
+import org.opensaml.saml.saml2.core.AuthnRequest;
+import org.opensaml.saml.saml2.core.NameID;
+import org.opensaml.saml.saml2.core.NameIDPolicy;
+import org.opensaml.saml.saml2.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This plugin is responsible for setting the <@code>NameID</@code>'s <@code>Format</@code> &
+ * <@code>SPNameQualifier</@code> on the outgoing <@code>Response</@code>. It does this by parsing
+ * the <@code>NameIDPolicy</@code> of the passed in <@code>AuthnRequest</@code>.
+ *
+ * <p>See section 3.4.1.1 of the SAML Core specification for more information.
+ */
+public class NameIdPolicyPresignPlugin implements SamlPresignPlugin {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(NameIdPolicyPresignPlugin.class);
+
+  @Override
+  public void processPresign(
+      Response response,
+      AuthnRequest authnRequest,
+      List<String> spMetadata,
+      Set<Binding> supportedBindings) {
+    NameIDPolicy nameIdPolicy = authnRequest.getNameIDPolicy();
+
+    if (nameIdPolicy == null
```

Comment:
❓ Why do this check separately instead of just filtering the stream below?

Reply:
It's a nice little short circuit for when we aren't given a `NameIDPolicy` in the `AuthnRequest`.